### ISSUE TITLE
[Inference] Reduce GPU memory usage for model

### ISSUE THREAD

**Issue by yuanxion** [yuanxion/Text2Video-Zero#2]
Colossal-AI may be used to reduce GPU Memory consumption during training (but may cause more training time).
https://github.com/hpcaitech/ColossalAI

Colossal-AI is used for acceleration of AIGC (AI-Generated Content) models such as [Stable Diffusion v1](https://github.com/CompVis/stable-diffusion) and [Stable Diffusion v2](https://github.com/Stability-AI/stablediffusion). (Nvidia GPU/Habana Gaudi)

**Comment by Xiangyi1996** [comment]
For inference, 
1. Colossal-AI achieves up to **1.42 times faster for single-GPU inference**
2. Reduce inference GPU memory consumption by 2.5x.

**Comment by yuanxion** [comment]
> For inference,
> 
> 1. Colossal-AI achieves up to **1.42 times faster for single-GPU inference**
> 2. Reduce inference GPU memory consumption by 2.5x.

Wow, that's great. Could you please create a PR for it, thx.

**Comment by yuanxion** [comment]
And maybe also collect the performance (baseline vs colossal-ai) data within a table, something like:

Performance | baseline | colossal-ai | deepspeed
-- | -- | -- | --
Running time (s) | | | 
GPU utilization (%) | | |
Memory usage (MB) | |  | 

Also, there is a `config.py` for this project, which may also helpful for saving GPU memory.

```
save_memory = False
// share.py
if config.save_memory:
    enable_sliced_attention()
```

**Comment by Xiangyi1996** [comment]
ColossalAI is a distributed deep learning framework that mainly targets the training and inference of large-scale models. It provides various parallelization techniques such as data parallelism, tensor parallelism and pipeline parallelism, as well as some optimization tools such as mixed precision training, gradient accumulation and offloading. For single-machine single-card models, ColossalAI may not have a significant effect, because these models do not require distributed systems or multi-card parallelism. If you want to use ColossalAI to accelerate single-machine single-card models, you can try to use its optimization tools, such as mixed precision training or gradient accumulation, to improve the training speed or increase the batch size. However, these tools can also be used in other frameworks, such as PyTorch or TensorFlow. Therefore, ColossalAI’s main advantage is on large-scale models, not on single-machine single-card models.

**Comment by Xiangyi1996** [comment]
# Introduction about Colossal AI
## Background
![Image](https://github.com/yuanxion/Text2Video-Zero/assets/22722992/941f3e78-bdf6-4624-9732-80a2a698251c)
## Parallelism
### Data Parallel
### ModelParallel
- Tensor Parallel
- Pipeline Parallel
## Sequence Parallelism
![Image](https://github.com/yuanxion/Text2Video-Zero/assets/22722992/b8e1f5a5-0bc3-49a4-8dd1-2c2c8b7fb178)
## Ring Self-attention


![Image](https://github.com/yuanxion/Text2Video-Zero/assets/22722992/ef1a9987-0a51-463a-8cde-620aadd25e62)

## Performance



![Image](https://github.com/yuanxion/Text2Video-Zero/assets/22722992/ba10b81a-5198-465a-8c03-0dfd420ee282)


## How to optimize the single-machine single-card model?

For single-machine single-card models, ColossalAI may not have a significant effect, because these models do not require distributed systems or multi-card parallelism. If you want to use ColossalAI to accelerate single-machine single-card models, you can try to use its optimization tools, such as mixed precision training or gradient accumulation, to improve the training speed or increase the batch size. However, these tools can also be used in other frameworks, such as PyTorch or TensorFlow. Therefore, ColossalAI’s main advantage is on large-scale models, not on single-machine single-card models.

**Comment by Xiangyi1996** [comment]
Performance | baseline | colossal-ai | deepspeed
-- | -- | -- | --
Running time (s) | 7 mins| | 
GPU utilization (%) | 99% | |
Memory usage (MB) | 9228MiB / 12288MiB|  |
